### PR TITLE
doc: add pull request and issue template and CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+<BLANK LINE>
+<footer>(optional)
+```
+
+The first line is the subject and should be no longer than 70 characters, the
+second line is always blank, and other lines should be wrapped at 80 characters.
+This allows the message to be easier to read on GitHub as well as in various
+git tools.
+
+If the change affects more than one subsystem, you can use comma to separate them like `server,pd-cilent:`.
```
Comment (lilin90) [inline CONTRIBUTING.md]: We leave a whitespace after the comma, so it should be `server, pd-cilent:`.

File: pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+Thank you for working on PD! Please read PD's [CONTRIBUTING](https://github.com/pingcap/pd/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
```
Comment (lilin90) [inline pull_request_template.md]: - It'd be better to put this file in `.github/pull_request_template.md`. 
- For this line, please put this in an HTML comment. 

Ref: 
- https://github.com/pingcap/tikv/pull/3154
- https://github.com/pingcap/tikv/blob/master/.github/pull_request_template.md

File: pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+## What are the type of the changes (required)?
+
+The currently defined types are listed below, please pick one of the types for this PR by removing the others:
+- New feature (non-breaking change which adds functionality)
+- Improvement (non-breaking change which is an improvement to an existing feature)
+- Bug fix (non-breaking change which fixes an issue)
+- Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+## How has this PR been tested (required)?
```
Comment (lilin90) [inline pull_request_template.md]: The word `required` is acceptable, but in other repos, the word `mandatory` is used. Change the several `required` to `mandatory` for consistency?

Comment (disksing) [reply pull_request_template.md]: I suppose `required` is more commonly used. I had looked up a dictionary to know what `mandatory` means...

**Comment by siddontang** [comment]
LGTM

Can you send an issue template too?

**Comment by disksing** [comment]
PTAL @lilin90

File: pull_request_template.md (hunk 3 above)
Comment (lilin90) [reply pull_request_template.md]: @disksing `required` is a synonym for `mandatory`. `mandatory` is generally used in the sense of "binding". Its usage is not limited to laws. For example, "mandatory training".

Comment (disksing) [reply pull_request_template.md]: IMO, _required_ is more commonly used to indicate whether a content must be filled in, contrast to _optional_. Here are some examples I can find:
- [HTML input required attribute](https://www.w3schools.com/TAgs/att_input_required.asp)
- [protocol buffers field rules](https://developers.google.com/protocol-buffers/docs/proto#specifying-field-rules)

I have also found a comparison between them on Quora: https://www.quora.com/Mandatory-compulsory-obligatory-essential-necessary-required-For-what-situations-are-each-if-these-words-best-suited 
Quoted below:
> Required suggests that a rule or procedure is operating. All visitors are required to wear a badge. Do you have to take the SAT? Yes, it's required if you're applying to college. 
> ...
> Mandatory means there is a mandate to perform a particular act in a particular situation. A flu shot is mandatory for all hospital staff. 

I suppose _required_ is more suitable to instruct that the questions must be answered in order to continue the pull request procedure.

Comment (lilin90) [reply pull_request_template.md]: Thanks for your attention to the choice of words! You can keep it.

**Pull request merged**